### PR TITLE
Add Image media bundle #483

### PR DESCRIPTION
**Pull request by jyobb**
I have added an image media bundle in the config/install directory.

### CODE REVIEW

**Comment by dannylamb** [comment]
This PR looks good @jyobb.  We'll need to add a field to hold the image contents.  Would you like to try and tackle that here?  I can help you out if you'd like.

If not, I'll peel off a separate ticket and we can have it addressed later.

**Comment by jyobb** [comment]
@dannylamb I would be glad to tackle adding the field to hold contents

**Comment by whikloj** [comment]
I'd agree that adding the configuration entity and the field definition/storage would be a good thing to add. Otherwise this work is hard to test. 

I mean it installed fine 👍 , just can't use it yet.

@jyobb  you might want to look at [islandora_collection](https://github.com/Islandora-CLAW/islandora_collection/tree/8.x-1.x/config/install) that is an entity with 2 fields defined and you can see the YAML files there.

**Comment by jyobb** [comment]
@whikloj @dannylamb Should I be able to just copy paste and tweak files from islandora_collection?

**Comment by whikloj** [comment]
@jyobb you might want to create the field using Drupal's UI and export the configuration instead. 
Only because this is the first field to store an image, those ones in islandora_collection were storing text and entity references.

**Comment by jyobb** [comment]
@whikloj I have created what I think is an image storage field but would like to run what I have done past you to make sure I'm not totally out to lunch.

**Comment by jyobb** [comment]
I'm not around Feb.20-24 so if this needs to be done right away someone will need to take it on.

**Comment by whikloj** [comment]
@jyobb Just throw what you have in here, don't worry about the mess. I have a strict _squash and merge_ policy, so those attempts will fade into the gloriousness of your success.

**Comment by whikloj** [comment]
@ruebot Can I get access to this repo por favor?

**Comment by dannylamb** [comment]
@jyobb It's no problem to wait 'til you get back.  Feel free to push up what you have when you can and we'll give it a whirl.

Thx so much for taking this on.  The D8 beast is a big one to wrestle with at first, and I appreciate the effort.

**Comment by dannylamb** [comment]
@jyobb Antyhing I can do to help you with this?

**Comment by jyobb** [comment]
I have a fedora Resource Image bundle but cannot see how to add the dependencies. If we could have a quick chat on IRC so I can understand how the yaml file is used that would be very helpful.

**Comment by dannylamb** [comment]
@jyobb The image and description fields have been made for `fedora_resource` entities.  But for here, all you need is to make the image field for `media` entities.  Specifically the `image` media entity you made with your first commit.  Don't worry about description or making a bundle for `fedora_resource`.

If you visit http://localhost:8000/admin/structure/media/manage/image/fields, you can add the image field there.  Once its added, you'll also want to configure the `image` bundle to point to it.  You can do so at the 'Type Provider Information' section at: http://localhost:8000/admin/structure/media/manage/image.

Re-export the bundle and then export the field and storage for the image.  Just those three files will do it for this PR.

**Comment by jyobb** [comment]
I have removed the Fedora Resource bundle yml files and added Islandora Image media bundle yml files

File: config/install/field.storage.media.field_image_storage.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+langcode: en
+status: true
+dependencies:
+  module:
+    - file
+    - media_entity
+id: media.field_image_storage
+field_name: field_image_storage
+entity_type: media
+type: file
```
Comment (dannylamb) [inline config/install/field.storage.media.field_image_storage.yml]: This needs to be an 'Image' field.  When you go to create the field, it's in the drop-down for field type.  It's pretty much the same as file, except it has some niceties for alt text and displaying the image.

File: config/install/field.field.media.image.field_image_storage.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+bundle: image
+label: 'Image Storage'
+description: ''
+required: false
+translatable: false
+default_value: {  }
+default_value_callback: ''
+settings:
+  file_directory: '[date:custom:Y]-[date:custom:m]'
+  file_extensions: 'png gif jpg jpeg tiff tif jp2'
```
Comment (dannylamb) [inline config/install/field.field.media.image.field_image_storage.yml]: I did some tests, and for some reason, the field isn't obeying these values.  I don't think it's anything you've done.  It's probably Drupal.  Let's leave the defaults (`png gif jpg jpeg`) until I can take a deeper dive and figure out what's going on.

**Review by dannylamb — CHANGES_REQUESTED** [review]

File: config/install/field.storage.media.field_image_storage.yml (hunk 1 above)
Comment (jyobb) [reply config/install/field.storage.media.field_image_storage.yml]: Ok. I'll fix it up.

File: config/install/field.field.media.image.field_image_storage.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+uuid: 1a7e8c9e-d18f-42a6-8e7c-632c9dde5aaf
+langcode: en
+status: true
+dependencies:
```
Comment (dannylamb) [inline config/install/field.field.media.image.field_image_storage.yml]: Oh, and I forgot.  You need to add a block that looks like this
```
enforced:
  module:
    - islandora_image
```
to the dependencies section of each of your exported yaml files.  This tells drupal to uninstall the config when you uninstall the module.  It makes testing a whole lot easier, too.

Here's an example from the `islandora` repo: https://github.com/Islandora-CLAW/islandora/blob/8.x-1.x/config/install/islandora.fedora_resource_type.rdf_source.yml#L5-L7

Comment (jyobb) [reply config/install/field.field.media.image.field_image_storage.yml]: is this done via the web UI? I hunted all over the place for a way to setup the dependencies but could not find anything

Comment (dannylamb) [reply config/install/field.field.media.image.field_image_storage.yml]: Sadly, no.  It's a manual process.  I add them in a text editor after exporting them.

File: config/install/field.field.media.image.field_islandora_image.yml [hunk 4]
```diff
@@ -7,6 +7,7 @@ dependencies:
     - media_entity.bundle.image
   module:
     - image
+    - islandora_image
```
Comment (dannylamb) [inline config/install/field.field.media.image.field_islandora_image.yml]: this dependency needs to be nested in a enforced block.  the entire chunk should look like this: 
```
dependencies:
  config:
    - field.storage.media.field_islandora_image
    - media_entity.bundle.image
  module:
    - image
  enforced:
    module:
      - islandora_image
```

**Review by dannylamb — CHANGES_REQUESTED** [review]

**Comment by jyobb** [comment]
@dannylamb I have added the dependency.

**Comment by jyobb** [comment]
@dannylamb I have corrected the dependency section of the yml files

**Comment by dannylamb** [comment]
@jyobb pulling these in now to test

**Comment by dannylamb** [comment]
:+1: Looks good.  Works as expected.

On behalf of CLAW, I'll let Michael Bolton congratulate you on your first contribution!

<image src="https://media.giphy.com/media/l3q2UdX8HREZi35gA/source.gif" width="480" height="270"></image>

**Pull request merged**